node/src/utils.rs

```diff
@@ -405,6 +407,47 @@ pub(crate) async fn wait_for_arc_drop<T>(
     false
 }
 
+/// A thread-safe wrapper around a file that writes chunks.
+///
+/// A chunk can (but needn't) be a line. The writer guarantees it will be written to the wrapped
+/// file, even if other threads are attempting to write chunks at the same time.
+#[derive(Clone)]
+pub(crate) struct LockedLineWriter(Arc<Mutex<File>>);
```

Comment:
Did you consider `tokio::fs::File` here? Seems like it already wraps a mutex like you're doing here. https://docs.rs/tokio/latest/src/tokio/fs/file.rs.html#93-96

In addition, you probably wouldn't have to handle lock poisoning as per below.

Reply:
The latest version of `tokio::fs::File` does not implement synchronous `io::Write`, as far as I can tell. We need a synchronous version though, since all of the writing is done in a sync callback that is passed to the OpenSSL C code, so I think we cannot get away with it here.